CONTRIBUTING.md

```diff
@@ -97,24 +97,38 @@ Mitiq code is developed according the best practices of Python development.
+and `make check-format` (to see if `Black` would reformat the code) in the top-level directory of
 the repository. If you aren't presented with any errors, then that means your code is good enough
-for the linter (`flake8`) and formatter (`black`). If `make check-format` fails, it will present
+for the linter (`flake8`) and formatter (`Black`). If `make check-format` fails, it will present
 you with a diff, which you can resolve by running `make format`. Black is very opinionated, but
 saves a lot of time by removing the need for style nitpicks in PR review. We only deviate from its
 default behavior in one category: we choose to use a line length of 79 rather than the Black
 default of 88 (this is configured in the [`pyproject.toml`](https://github.com/unitaryfund/mitiq/blob/master/pyproject.toml) file).
 
+#### Some possible issues with `Black`
+Below is a summarized list of some issues leading to errors. A more detailed discussion can be found [in the documentation](https://black.readthedocs.io/en/stable/the_black_code_style.html#the-black-code-style).
+- In general, `Black` strives for one expression or statement per line with uniform vertical and horizontal whitespace. This is described as `Black` doing everything to keep "[`pycodestyle` happy](https://black.readthedocs.io/en/stable/the_black_code_style.html#how-black-wraps-lines)". A list of errors due to `pycodestyle` can be found [here](https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes).
+- When a statement or expression is moved to its own line, a [trailing comma](https://black.readthedocs.io/en/stable/the_black_code_style.html#trailing-commas) will be added.
+- Because uniform vertical whitespace is enforced, empty lines are only allowed in a [handful of scenarios](https://black.readthedocs.io/en/stable/the_black_code_style.html#empty-lines) like inside, before and after function definitions etc.
+- [Default line length](https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length) for `Black` is 88 characters per line. This number can be adjusted to have fewer or more characters as needed. As stated above, default length for `Mitiq` is 79 characters.
+- If an expression or a statement can be fit in one line, trailing comma introduced before a line break will be removed along with the [parentheses](https://black.readthedocs.io/en/stable/the_black_code_style.html#parentheses).
+- If backslashes are used for line breaks, these will be replaced with parentheses in keeping with [PEP 8 style guide](https://www.python.org/dev/peps/pep-0008/#maximum-line-length).
+- To obey [PEP 8 style guide for improving readability](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator), a line break will be introduced before a binary operator.
+- Use double quotes for [strings](https://black.readthedocs.io/en/stable/the_black_code_style.html#strings).
```

Comment:
I think we should just link to the black docs instead of writing all of this text here since (1) then we never have to update this and (2) the point of using black is not having to think about all these formatting things.

Reply:
> subsection making the existing text more evident

Not sure if I misread the issue but this was stated as one of the additions in the issue.

Reply:
I'll leave the decision to @nathanshammah then.

Reply:
Crushing these issues @purva-thakre! Nice 💯 ! Would it make sense here to format _Black_ in italics and reserve `black` for code reference uses?

Reply:
I agree with @rmlarose comment and also with @Aaron-Robertson suggestion.

Reply:
I'll remove this section and add suggested code block for `black` format. I did misread the issue @rmlarose  !